src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -114,25 +115,58 @@ const UsersFragment = ({
     if (searchText.length > 0) fetchData();
   };
 
-  const renderRecentSearch = (index) => {
-    return (
-      index === 0 &&
-      !withoutRecent && (
-        <RecentSearch
-          shown={showRecentSearch || isFirstTimeOpen}
-          setSearchText={setSearchText}
-          setIsFirstTimeOpen={setIsFirstTimeOpen}
+  const SECTIONS = [
+    {
+      title: 'First',
+      content: 'Lorem ipsum...'
+    }
+  ];
+
+  const AccordionView = ({data}) => {
+    const [activeSections, setActiveSections] = useState([]);
+
+    const renderSectionTitle = () => {
+      return <View style={styles.content}></View>;
+    };
+
+    const renderHeader = (data, index) => {
+      return (
+        <DiscoveryTitleSeparator
+          key="user-title-separator"
+          text="People you follow"
+          showArrow
+          rotateArrow={activeSections?.some((actived) => actived === index)}
         />
-      )
+      );
+    };
+
+    const renderContent = () => {
+      return (
+        <View style={styles.content}>{data?.map((item, index) => renderItem({index, item}))}</View>
+      );
+    };
+
+    const updateSections = (activeSectionsParams) => {
+      setActiveSections(activeSectionsParams);
+    };
+
+    return (
+      <Accordion
+        sections={SECTIONS}
+        activeSections={activeSections}
+        renderSectionTitle={renderSectionTitle}
+        renderHeader={renderHeader}
+        renderContent={renderContent}
+        onChange={updateSections}
+      />
     );
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `AccordionView` component is defined inside another function. This could lead to unnecessary re-renders and performance issues as the `AccordionView` component will be redefined every time the parent function is executed. It would be better to define this component outside of the parent function and pass any necessary data via props.

```diff
-  const AccordionView = ({data}) => {
+  const AccordionView = React.memo(({data}) => {
     const [activeSections, setActiveSections] = useState([]);

     const renderSectionTitle = () => {
       return <View style={styles.content}></View>;
     };

     const renderHeader = (data, index) => {
       return (
         <DiscoveryTitleSeparator
           key="user-title-separator"
           text="People you follow"
           showArrow
           rotateArrow={activeSections?.some((actived) => actived === index)}
          />
       );
     };

     const renderContent = () => {
       return (
         <View style={styles.content}>{data?.map((item, index) => renderItem({index, item}))}</View>
       );
     };

     const updateSections = (activeSectionsParams) => {
       setActiveSections(activeSectionsParams);
     };

     return (
       <Accordion
         sections={SECTIONS}
         activeSections={activeSections}
         renderSectionTitle={renderSectionTitle}
         renderHeader={renderHeader}
         renderContent={renderContent}
         onChange={updateSections}
       />
      );
-    };
+    }, (prevProps, nextProps) => prevProps.data === nextProps.data);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->